tests/src/test/scala/it/unibo/scafi/test/functional/stdlib/FieldUtils.scala

```diff
@@ -0,0 +1,205 @@
+      )
+    }, ntimes = fewRounds)(net)
+    net.chgSensorValue("sensorZero", Set(0), false)
+
+    //ASSERT
+    assertNetworkValues((0 to 8).zip(List(
+      (false, true), (true, true), (false, false),
+      (true, true), (true, true), (false, false),
+      (false, false), (false, false), (false, false)
+    )).toMap)(net)
+  }
+
+  Field_Utils should "support anyHood, more neighbours with true value" in new SimulationContextFixture {
+    // ACT
+    net.addSensor[Boolean](name = "sensorZero", value = false)
+    net.chgSensorValue("sensorZero", Set(0,3), true)
+    exec(new TestProgram {
+      override def main(): Any = (
+        excludingSelf.anyHood(nbr(sense[Boolean]("sensorZero"))),
+        includingSelf.anyHood(nbr(sense[Boolean]("sensorZero")))
+      )
+    }, ntimes = fewRounds)(net)
+    net.chgSensorValue("sensorZero", Set(0, 3), false)
+
+    //ASSERT
+    assertNetworkValues((0 to 8).zip(List(
+      (true, true), (true, true), (false, false),
+      (true, true), (true, true), (false, false),
+      (true, true), (true, true), (false, false)
+    )).toMap)(net)
+  }
+  
+  Field_Utils should "support everyHood" in new SimulationContextFixture {
+    // ACT
+    exec(new TestProgram {
+      override def main(): Any = (
+        excludingSelf.everyHood(mid() == nbr(mid())),
+        includingSelf.everyHood(mid() == nbr(mid()))
+      )
+    }, ntimes = fewRounds)(net)
+
+    //ASSERT
+    /*
+      * def everyHood(expr: => Boolean): Boolean = foldhoodTemplate(true)(_&&_)(expr)
+      * everyHood is initialized to true.
+      * Since the last device is alone it has no neighborhood no fold operation is performed
+      *   resulting in a final true vale
+      */
+    assertNetworkValues((0 to 8).zip(List(
+      (false, false), (false, false), (false, false),
+      (false, false), (false, false), (false, false),
+      (false, false), (false, false), (true, true)
+    )).toMap)(net)
+  }
+
+  Field_Utils should "support sumHood" in new SimulationContextFixture {
+    // ACT
+    exec(new TestProgram {
+      override def main(): Any = (
+        excludingSelf.sumHood(1),
+        includingSelf.sumHood(1)
+      )
+    }, ntimes = fewRounds)(net)
+
+    //ASSERT
+    assertNetworkValues((0 to 8).zip(List(
+      (net.neighbourhood(0).size, net.neighbourhood(0).size + 1), (net.neighbourhood(1).size, net.neighbourhood(1).size + 1), (net.neighbourhood(2).size, net.neighbourhood(2).size + 1),
+      (net.neighbourhood(3).size, net.neighbourhood(3).size + 1), (net.neighbourhood(4).size, net.neighbourhood(4).size + 1), (net.neighbourhood(5).size, net.neighbourhood(5).size + 1),
+      (net.neighbourhood(6).size, net.neighbourhood(6).size + 1), (net.neighbourhood(7).size, net.neighbourhood(7).size + 1), (net.neighbourhood(8).size, net.neighbourhood(8).size + 1)
+    )).toMap)(net)
+  }
+
+  Field_Utils should "support sumHood - mid" in new SimulationContextFixture {
```

Comment:
This test case is complicated with no apparent reason.
Why don't you just use an input network where each device has a sensor that provide a small value (e.g., between -10 and 10) and manually compute the result.
By the way, we could also test adaptivity to value changes..